### PR TITLE
feat: improve Nuxt3 compatibility

### LINKED ISSUE #216: Nuxt 3: `plugin:vue/vue3-recommended` instead of `plugin:vue/recommended`
Is this the right package to use with ESLint for Nuxt 3 development? I keep seeing Vue 2-related errors which are most likely due to the following rule `plugin:vue/recommended`.

### PR DESCRIPTION
**Pull request by antfu**
close #231
close #216

### CODE REVIEW

**Comment by Atinux** [comment]
Ping us when ready @antfu :)

**Comment by antfu** [comment]
It's ready

File: packages/eslint-config/index.js [hunk 1]
```diff
@@ -1,13 +1,22 @@
+const { getPackageInfoSync } = require('local-pkg')
+
+const nuxt = getPackageInfoSync('nuxt')
+const isNuxt2 = nuxt && nuxt.version && nuxt.version.startsWith('2.')
+// console.log({ isNuxt2 })
```
Comment (DamianGlowala) [inline packages/eslint-config/index.js:5]:
```suggestion
```

Comment (antfu) [reply packages/eslint-config/index.js:5]: I actually left it intentional but no strong opinion to keep it either.

**Review by clarkdo — APPROVED** [review]

**Comment by gjfei** [comment]
image
image
image

doesn't seem to work

**Comment by regchiu** [comment]
@antfu

Excuse me, Continuing to use vue 2's vue/no-multiple-template-root rule because of this Nuxt rule?
Thanks.
https://nuxt.com/docs/guide/directory-structure/pages/#usage

image

**Pull request merged**